platforms/ide/problems-rendering/src/test/groovy/org/gradle/problems/internal/rendering/ProblemRendererTest.groovy

```diff
@@ -128,10 +130,11 @@ class ProblemRendererTest extends Specification {
 
         then:
         renderedText.normalize() == """\
-            |  display-name
+            |display-name
+            |  Problem details:
             |    details:1
             |    details:2
-            |  display-name
+            |  Some context for one problem
             |    details:1
             |    details:2""".stripMargin()
```

Comment:
:x: Hmm, I don't think `ProblemId` should be used to group up problems. IMO they should be displayed per problem, especially as they carry the fallback information (the display name) when the `contextualLabel` is missing. Instead, only `ProblemGroup` should be used for the hierarchical grouping.

How about this format?
```
level1Group:
  display-name: // This problem doesn't have a `contextualLabel`, so we show the `ProblemId.displayName`
    details:1
    details:2
  Some context for one problem // This problem has a contextual label, so we show that instead
    details:1
    details:2
level2Group:
  display-name:
     ... // So on, so forth
```

Reply:
so `ProblemGroup` is a potential hierarchy, that makes total sense to use as a way to organize problems in a full report.
But I feel it is too far removed from a problem tied to an exception. Since the API really requires a `ProblemId` when interacting with problems, I think that's the right level to focus on here.
It also gives a clear purpose to that `ProblemId.displayName`.

Hiding the problemId inherently forces me to duplicate information when creating my problems.
I don't think we should design a rendering where one piece of information hides the other when provided. It should be a strict hierarchy.